test/integration/bad_key_test.go

```diff
@@ -0,0 +1,125 @@
+//go:build integration
+
+package integration
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/eggsampler/acme/v3"
+	"github.com/letsencrypt/boulder/test"
+)
+
+// TestFermat ensures that a certificate public key which can be factored using
+// less than 100 rounds of Fermat's Algorithm is rejected.
+func TestFermat(t *testing.T) {
+	t.Parallel()
+
+	// Create a base account to use for revocation tests.
+	os.Setenv("DIRECTORY", "http://boulder.service.consul:4001/directory")
+
+	type testCase struct {
+		name string
+		p    string
+		q    string
+	}
+
+	testCases := []testCase{
+		{
+			name: "canon printer (2048 bit, 1 round)",
+			p:    "155536235030272749691472293262418471207550926406427515178205576891522284497518443889075039382254334975506248481615035474816604875321501901699955105345417152355947783063521554077194367454070647740704883461064399268622437721385112646454393005862535727615809073410746393326688230040267160616554768771412289114449",
+			q:    "155536235030272749691472293262418471207550926406427515178205576891522284497518443889075039382254334975506248481615035474816604875321501901699955105345417152355947783063521554077194367454070647740704883461064399268622437721385112646454393005862535727615809073410746393326688230040267160616554768771412289114113",
+		},
+		{
+			name: "innsbruck printer (4096 bit, 1 round)",
+			p:    "25868808535211632564072019392873831934145242707953960515208595626279836366691068618582894100813803673421320899654654938470888358089618966238341690624345530870988951109006149164192566967552401505863871260691612081236189439839963332690997129144163260418447718577834226720411404568398865166471102885763673744513186211985402019037772108416694793355840983833695882936201196462579254234744648546792097397517107797153785052856301942321429858537224127598198913168345965493941246097657533085617002572245972336841716321849601971924830462771411171570422802773095537171762650402420866468579928479284978914972383512240254605625661",
+			q:    "25868808535211632564072019392873831934145242707953960515208595626279836366691068618582894100813803673421320899654654938470888358089618966238341690624345530870988951109006149164192566967552401505863871260691612081236189439839963332690997129144163260418447718577834226720411404568398865166471102885763673744513186211985402019037772108416694793355840983833695882936201196462579254234744648546792097397517107797153785052856301942321429858537224127598198913168345965493941246097657533085617002572245972336841716321849601971924830462771411171570422802773095537171762650402420866468579928479284978914972383512240254605624819",
+		},
+		// Ideally we'd have a 2408-bit, nearly-100-rounds test case, but it turns
+		// out purposefully generating keys that require 1 < N < 100 rounds to be
+		// factored is surprisingly tricky.
+	}
+
+	// Try to issue a certificate with
```

Comment:
```suggestion
	// Try to issue a certificate with bunny ears
```

or

```suggestion
	// Try to issue a certificate with panache
```

or

```suggestion
	// Try to issue a certificate without using a computer
```

?

Reply:
I like the first one 👍🏻